tests/comptroller/comptroller.test.ts

```diff
@@ -0,0 +1,779 @@
+    handleNewMints([newMintEvent3, newMintEvent4])
+    assert.entityCount("MintEvent", 4) // There should be four mints now
+    assert.fieldEquals("MintEvent", mintID3, "id", mintID3)
+    assert.fieldEquals("MintEvent", mintID3, "to", testAccountAddress1)
+    assert.fieldEquals("MintEvent", mintID3, "from", xcksmMarketAddress)
+    assert.fieldEquals("MintEvent", mintID3, "amount", expectedAmount3.toString())
+    assert.fieldEquals("MintEvent", mintID4, "id", mintID4)
+    assert.fieldEquals("MintEvent", mintID4, "to", testAccountAddress2)
+    assert.fieldEquals("MintEvent", mintID4, "from", xcksmMarketAddress)
+    assert.fieldEquals("MintEvent", mintID4, "amount", expectedAmount4.toString())
+  })
+
+  test("Handle a Mint-type Transfer event (market -> account)", () => {
+    let transferAmount = BigInt.fromString("100000000000000000000")
+    // Need to mock the balanceOf function for the from account (the market contract)
+    createMockedFunction(
+      Address.fromString(movrMarketAddress),
+      "balanceOf", "balanceOf(address):(uint256)",
+    )
+      .withArgs([ethereum.Value.fromAddress(Address.fromString(movrMarketAddress))])
+      .returns([ethereum.Value.fromUnsignedBigInt(zeroBI)])
+    // Also need to mock the balanceOf function for the to account (test account 1)
+    createMockedFunction(
+      Address.fromString(movrMarketAddress),
+      "balanceOf", "balanceOf(address):(uint256)",
+    )
+      .withArgs([ethereum.Value.fromAddress(Address.fromString(testAccountAddress1))])
+      .returns([ethereum.Value.fromUnsignedBigInt(transferAmount)])
+
+    // Call mappings
+    let newHashString = "0x" + "1".repeat(64) // Generate a unique transaction hash
+    let newTransferEvent = createNewTransferEvent(
+      movrMarketAddress,
+      movrMarketAddress, // Simulate a mint from the market contract to test account 1
+      testAccountAddress1,
+      transferAmount,
+      newHashString,
+    )
+
+    let transferID = newHashString + "-1" // Transfer ID is the transaction hash + '-' + transaction.logIndex which is always 1
+    assert.fieldEquals("Market", movrMarketAddress, "supplierCount", "0") // There should be no suppliers yet
+    assert.entityCount("TransferEvent", 0) // There should be no transfers yet
+    handleNewTransfers([newTransferEvent])
+    assert.entityCount("TransferEvent", 1) // There should be one transfer now
+    assert.fieldEquals("Market", movrMarketAddress, "supplierCount", "1") // There should be one supplier now
+    assert.fieldEquals("TransferEvent", transferID, "id", transferID)
+    assert.fieldEquals("TransferEvent", transferID, "from", movrMarketAddress)
+    assert.fieldEquals("TransferEvent", transferID, "to", testAccountAddress1)
+    assert.fieldEquals("TransferEvent", transferID, "amount", transferAmount.toBigDecimal().div(cTokenDecimalsBD).toString())
+  })
+
+  test("Handle a Liquidation-type 1 (seize mTokens) Transfer event (liquidated -> liquidator)", () => {
```

Comment:
This is the test case that will both decrement `supplierCount` and increment `supplierCount` as the cTokens are transferred between two EOA accounts. (see above)